great_expectations/rule_based_profiler/parameter_builder/regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,193 @@
+        string exceeds the configured threshold.
+
+        :return: ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and optional details
+        """
+        validator: Optional[Validator] = self.get_validator(
+            domain=domain, variables=variables, parameters=parameters
+        )
+        if not validator:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"{self.__class__.__name__} was not able to get Validator using domain, variables and parameters provided."
+            )
+
+        batch_ids: Optional[List[str]] = self.get_batch_ids(
+            domain=domain, variables=variables, parameters=parameters
+        )
+        if not batch_ids:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"Utilizing a {self.__class__.__name__} requires a non-empty list of batch identifiers."
+            )
+
+        metric_computation_result: MetricComputationResult
+
+        metric_values: np.ndarray
+
+        metric_computation_result: MetricComputationResult = self.get_metrics(
+            batch_ids=batch_ids,
+            validator=validator,
+            metric_name="column_values.nonnull.count",
+            metric_domain_kwargs=self._metric_domain_kwargs,
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+        metric_values = metric_computation_result.metric_values
+        # Now obtain 1-dimensional vector of values of computed metric (each element corresponds to a Batch ID).
+        metric_values = metric_values[:, 0]
+
+        nonnull_count: int = sum(metric_values)
+
+        regex_string_success_ratios: dict = {}
+
+        regex_string: str
+        match_regex_metric_value_kwargs: dict
+        for regex_string in self._candidate_regexes:
+            if self._metric_value_kwargs:
+                match_regex_metric_value_kwargs: dict = {
+                    **self._metric_value_kwargs,
+                    **{"regex": regex_string},
+                }
+            else:
+                match_regex_metric_value_kwargs: dict = {"regex": regex_string}
+
+            metric_computation_result: MetricComputationResult = self.get_metrics(
+                batch_ids=batch_ids,
+                validator=validator,
+                metric_name="column_values.match_regex.unexpected_count",
+                metric_domain_kwargs=self._metric_domain_kwargs,
+                metric_value_kwargs=match_regex_metric_value_kwargs,
+                domain=domain,
+                variables=variables,
+                parameters=parameters,
+            )
+            metric_values = metric_computation_result.metric_values
+            # Now obtain 1-dimensional vector of values of computed metric (each element corresponds to a Batch ID).
+            metric_values = metric_values[:, 0]
```

Comment:
@Shinnnyshinshin If needed, please see the latest way of converting N-dimensions to 1 dimension in `MetricMultiBatchParameterBuilder` -- thanks.